docker-compose.yml

```diff
@@ -35,6 +35,24 @@ services:
       - shared_audits:/var/geoserver/audits:rw
       - ./config/context.xml:/usr/local/tomcat/conf/context.xml:ro
 
+  haproxy:
```

Comment:
we've never used Haproy for this purpose, only Nginx.
We discussed it extensively over skype

Reply:
@randomorder 
I have used HAProxy as a solution (2 min solution, since I have used before) for the purpose of testing  GS config promotion script pushed with this PR (file: jenkins-catalog-promotion.sh). Adding HAProxy allowed me to create as many slave instances as desired, and then running the mentioned sh it's using the docker container names as a convention to filter list of all the slaves instances that should be restarted.
It's an alternative that could resolve the proxy issue (you already mentioned the other options, I'm ok with proceeding with any of them. You decide which one).

Reply:
> @randomorder
> I have used HAProxy as a solution (2 min solution, since I have used before) for the purpose of testing GS config promotion script pushed with this PR (file: jenkins-catalog-promotion.sh). Adding HAProxy allowed me to create as many slave instances as desired, and then running the mentioned sh it's using the docker container names as a convention to filter list of all the slaves instances that should be restarted.

Ok I guess it ended up in the PR for review by mistake

> It's an alternative that could resolve the proxy issue (you already mentioned the other options, I'm ok with proceeding with any of them. You decide which one).

Nginx reverse proxy then

Reply:
> Nginx reverse proxy then

ok. After documentation task I will check how to do this.